app/controllers/webhook_controller.rb

```diff
@@ -31,6 +31,8 @@ def callback
           text = fetch_text_in_image(message_content)
         when Line::Bot::Event::MessageType::Audio
           text = fetch_text_in_audio(message_content)
+        when Line::Bot::Event::MessageType::File
+          text = fetch_text_in_file(event.message['fileName'], message_content)
```

Comment:
[imo] 個人的には拡張子バリデーション→ fetch_text_in_audioの順で分けて処理したいですね。包含するのではなく。
具体的には
```suggestion
          return NOT_AVAILABLE_FORMAT if validate_audio_format(event.message['fileName'])
          fetch_text_in_file(message_content)
```

というような形にして

```ruby
  def validate_audio_format(file_name)
    AVAILABLE_AUDIO_FILE_FORMATS.include?(File.extname(file_name))
  end
```

みたいなバリデータを入れる形？（なんかあんましだなぁ・・・）

目的としては以下の通りです。
1. とにかく早くエラーを返してあげる。
2. 一つの関数が持つ責務を一つに抑える

Reply:
了解です！